dbms/src/DataStreams/FilterBlockInputStream.h

```diff
@@ -42,7 +45,25 @@ class FilterBlockInputStream : public IProfilingBlockInputStream
     Block getHeader() const override;
 
 protected:
-    Block readImpl() override;
+    Block readImpl() override
+    {
+        // FilterPtr filter_ignored;
+        // return readImpl(filter_ignored, false);
+        // TODO: for test, remove it after test
```

Comment:
I'm a bit confused about this line.. Can you add some comments to explain it?

Reply:
Will it affect the performance of the filter?

Reply:
```cpp
FilterPtr filter = nullptr;
Block res = readImpl(filter, true);
if (filter && res)
{
   size_t passed_rows = countBytesInFilter(*filter);
   for (auto & col : res)
   {
       col.column = col.column->filter(*filter, passed_rows);
   }
}
return res;
```
It is just for test, I will remove after review. And use the following code.

```cpp
FilterPtr filter_ignored;
return readImpl(filter_ignored, false);
```

Reply:
> Will it affect the performance of the filter?

A little bit regression. 

before

```cpp
filter = const_cast<IColumn::Filter *>(filter_and_holder.data);
filter_holder = filter_and_holder.data_holder;
```

now

```cpp
filter.resize(rows);
std::copy(filter_and_holder.data->cbegin(), filter_and_holder.data->cend(), filter.begin());
```

A extra vector copy.

Reply:
Add `ColumnPtr filter_holder` in `FilterTransformAction` to hold data in `FilterDescription`, the vector copy is reduced.